src/test/java/com/imagination/cbs/service/impl/Html2PdfServiceImplTest.java

```diff
@@ -0,0 +1,137 @@
+package com.imagination.cbs.service.impl;
+
+import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.sql.Date;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import com.imagination.cbs.domain.Booking;
+import com.imagination.cbs.domain.BookingRevision;
+import com.imagination.cbs.domain.BookingWorkTask;
+import com.imagination.cbs.domain.Contractor;
+import com.imagination.cbs.domain.Discipline;
+import com.imagination.cbs.domain.RoleDm;
+
+import freemarker.core.ParseException;
+import freemarker.template.Configuration;
+import freemarker.template.MalformedTemplateNameException;
+import freemarker.template.Template;
+import freemarker.template.TemplateNotFoundException;
+
+@RunWith(MockitoJUnitRunner.class)
+public class Html2PdfServiceImplTest {
+
+	@InjectMocks
+	Html2PdfServiceImpl html2PdfServiceImpl;
+	
+	@Mock
+	private Configuration config;
+	
+	private Template template;
+
+	@Before
+	public void init() throws TemplateNotFoundException, MalformedTemplateNameException, ParseException, IOException {
+		template = mock(Template.class);
+	}
+	
+	@Test
+	public void shouldReturnNullByteArrayOutputStreamWhenExceptionOccured() throws TemplateNotFoundException, MalformedTemplateNameException, ParseException, IOException {
+		
+		final BookingRevision latestBookingRevision = createBookingRevision(false);
+		
+		when(config.getTemplate("pdf.agreement.ftl")).thenReturn(template);
+		
+		ByteArrayOutputStream byteArrayOutputStream = html2PdfServiceImpl.generateAgreementPdf(latestBookingRevision);
+		
+		verify(config).getTemplate("pdf.agreement.ftl");
+		
+		assertNull(byteArrayOutputStream);
+	}
+
+	@Test
+	public void shouldReturnNullByteArrayOutputStreamWhenExceptionOccuredWithEmptyValuesInBookingRevision() throws TemplateNotFoundException, MalformedTemplateNameException, ParseException, IOException {
+		
+		final BookingRevision latestBookingRevision = createBookingRevision(true);
+		
+		when(config.getTemplate("pdf.agreement.ftl")).thenReturn(template);
+		
+		ByteArrayOutputStream byteArrayOutputStream = html2PdfServiceImpl.generateAgreementPdf(latestBookingRevision);
+		
+		verify(config).getTemplate("pdf.agreement.ftl");
+		
+		assertNull(byteArrayOutputStream);
+	}
+
+	private BookingRevision createBookingRevision(boolean isEmpty) {
```

Comment:
this method is big, try to extract the code in common private methods, like creating role etc.

Reply:
well we dont have checkstyle here else it might complaint about method is big.

Reply:
Did changes according to suggestion